File: Programing_Basics_with_JS/conditionalStatementsAdvancedMoreExercises/08.PointOnRectangleBorder.js
function pointOnReactangleBorder(input) {
    let x1 = Number(input[0]);
    let y1 = Number(input[1]);
    let x2 = Number(input[2]);
    let y2 = Number(input[3]);
    let x = Number(input[4]);
    let y = Number(input[5]);

    if (x === x1 || x === x2) {
        if (y >= y1 && y <= y2) {
            console.log('Border');
        } else {
            console.log('Inside / Outside');
        }
    } else if (y === y1 || y === y2) {
        if (x >= x1 && x <= x2) {
            console.log('Border');
        } else {
            console.log('Inside / Outside');
        }
    } else {
        console.log('Inside / Outside');
    }
}

pointOnReactangleBorder(['2', '-3', '12', '3', '8', '-1'])
pointOnReactangleBorder(['2', '-3', '12', '3', '12', '-1'])
pointOnReactangleBorder(['2', '-3', '12', '3', '2', '4'])
pointOnReactangleBorder(['2', '-3', '12', '3', '13.456', '3'])

File: Programming_Basics_with_Python/conditional_statements_advanced_more_exercises/08._point_on_rectangle_border.py
x1 = float(input())
y1 = float(input())
x2 = float(input())
y2 = float(input())
x = float(input())
y = float(input())

message = ''
if x == x1 or x == x2:
    if y1 <= y <= y2:
        message = 'Border'
    else:
        message = 'Inside / Outside'
elif y == y1 or y == y2:
    if x1 <= x <= x2:
        message = 'Border'
    else:
        message = 'Inside / Outside'
else:
    message = 'Inside / Outside'

print(message)